### DISCUSSION TITLE
“Shadow image” of prompt message of the image-keyboard-response plugin

### DISCUSSION THREAD

**Discussion by ontinglo** [jspsych/jsPsych#1010] [Q&A] [answered]
Hi guys

I've a problem regarding the prompt message when using the jspsych-image-keyboard-response plugin.

I don't know why the prompt message will display at the same time(at the center of screen) with the stimulus photo together. Although the prompt message will be displayed under the stimulus photo ultimately, it makes the prompt message look like a shadow image when the experiment is running. Please see the photo in the following
![91145381-f32f0400-e6e7-11ea-8209-bd9c64d2f52f](https://user-images.githubusercontent.com/68586579/91383358-8ab66300-e85e-11ea-807f-a04cc35a41e5.jpeg)
The prompt message is displayed at the same time with the stimulus photo at the center of screen.

![91145462-122d9600-e6e8-11ea-89bc-e0c3c9d5b60e](https://user-images.githubusercontent.com/68586579/91383345-7c684700-e85e-11ea-94db-77ba21bba8f7.jpeg)
After that, the prompt message will be displayed normally under the stimulus photo.

![91145605-4f922380-e6e8-11ea-8635-d19e1eddcabc](https://user-images.githubusercontent.com/68586579/91383368-8f7b1700-e85e-11ea-8b4e-6914fba27c8f.jpeg)
I think the codes are pretty standard?

Could you please help me.
Thanks a lot!

**Comment by becky-gilbert** [comment, marked as answer]
Hi @ontinglo, I'm not totally sure I understand what's happening here. Is it the case that the prompt appears under higher up on the screen while the image is loading (your first screenshot), and then once the image has loaded it moves down the screen (your second screenshot)? Are you preloading the images?

**Reply by becky-gilbert** [reply]
p.s. it would be helpful if you could paste any code into your comment, rather than showing it in a screenshot. This will make it easier for us to test your code. You can create a code block using three backticks before and after the code, like this:
````
```
<paste your code here>
```
````
And if you write the word javascript after the first three backticks, that will give you JavaScript syntax highlighting.
````
```javascript
<paste your code here>
```
````

**Reply by ontinglo** [reply]
> Hi @ontinglo, I'm not totally sure I understand what's happening here. Is it the case that the prompt appears under higher up on the screen while the image is loading (your first screenshot), and then once the image has loaded it moves down the screen (your second screenshot)? Are you preloading the images?

hi becky @becky-gilbert , yes this is exactly what i experienced while running the experiment

**Reply by ontinglo** [reply]
```javascript
//'trial_faceimg' is defined in another .js file to indicate each face images in that file
   var trial_faceimg = {
        type: 'image-keyboard-response',
        stimulus: jsPsych.timelineVariable('trial_faceimg'),
        choices: ['1', '2', '3', '4', '5', '6', '7', '8', '9'],
        prompt: '<p><strong>Use numeric keys(1-9) on top of your keyboard</strong> to <strong><u>indicate how trustworthy the face is (1:least; 9:most)</u></strong>.</p>'
    }

//there are 4 variables storing information regarding different sets of images that will be used in the experiment, all are preloaded within the jsPsych.init
  jsPsych.init({
        timeline: timeline,
        preload_images: all_ExptPrimeStim,
        preload_images: all_ExptFaceStim,
        preload_images: all_trial_ExptFaceStim,
        preload_images: all_trial_ExptPrimeStim,
        show_preload_progress_bar: true,
        show_progress_bar: true,
        exclusions: {
            min_width: 800,
            min_height: 600
        },
        default_iti: 50,
        on_finish: function() {
            jsPsych.data.displayData('csv');
        },
    })
```

**Reply by becky-gilbert** [reply]
I think the problem here is that, in your `jsPsych.init` function, you have multiple identical object keys (i.e. settings parameters) called `preload_images`. In JavaScript objects, each key has to be unique, and if you provide multiple identical keys then only the last key-value pair will be used. This means that, in your case, only the `preload_images: all_trial_ExptPrimeStim` key-value pair was used, which means only the images in that array were preloaded.

You can fix this by combining all of your images into a single array and using that array for the `preload_images` parameter inside of `jsPsych.init`. jsPsych does allow you to use a nested array (i.e. an array made up of more arrays) with `preload_images`. Here's an example:
```javascript
jsPsych.init({
  timeline: timeline,
  preload_images: [all_ExptPrimeStim, all_ExptFaceStim, all_trial_ExptFaceStim, all_trial_ExptPrimeStim],
  show_preload_progress_bar: true,
  show_progress_bar: true,
  exclusions: {
    min_width: 800,
    min_height: 600
  },
  default_iti: 50,
  on_finish: function() {
    jsPsych.data.displayData('csv');
  },
})
```
Could you try this and see if it fixes the problem?

**Reply by ontinglo** [reply]
@becky-gilbert it works!! thanks so much!
